src/Validation/Factory.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation;
+
+/**
+ * Factory for creating restrictions.
+ */
+class Factory {
+	// COMBINATORS
+
+	/**
+	 * Get a restriction that sequentially checks the supplied restrictions.
+	 *
+	 * The new restriction tells the problem of the first violated restriction.
+	 *
+	 * @param   Restriction[]   $others
+	 * @return  Restriction
+	 */
+	public function seq(array $others);
+
+	/**
+	 * Get a restriction that checks the supplied restrictions in parallel.
+	 *
+	 * The new restriction tells the problems of all violated restrictions.
+	 *
+	 * @param   Restriction[]   $others
+	 * @return	Restriction
+	 */
+	public function par(array $others);
```

Comment:
Matter of personal taste: I hate those 3 letter Abbreviation. Here the problem is bigger than above. par to me those not immediatly hint for parellel. Could also be parity or similar.